lib/number/si.ex

```diff
@@ -0,0 +1,110 @@
+defmodule Number.SI do
+  @moduledoc """
+  Provides functions for formatting numbers using SI notation.
+  """
+
+  @prefixes [
+    {8,  "Y"},   # yotta
+    {7,  "Z"},   # zetta
+    {6,  "E"},   # exa
+    {5,  "P"},   # peta
+    {4,  "T"},   # tera
+    {3,  "G"},   # giga
+    {2,  "M"},   # mega
+    {1,  "k"},   # kilo
+    {0,  ""},
+    {-1, "m"},   # milli
+    {-2, "µ"},   # micro
+    {-3, "n"},   # nano
+    {-4, "p"},   # pico
+    {-5, "f"},   # femto
+    {-6, "a"},   # atto
+    {-7, "z"},   # zepto
+    {-8, "y"}    # ycoto
+  ]
+
+  @doc """
+  Format numbers using SI notation
+
+  ## Parameters
+
+  * `number` - A value to convert. Can be any value that implements
+    `Number.Conversion.to_float/1`.
+
+  * `options` - A keyword list of options. See the documentation below for all
+    available options.
+
+  ## Options
+
+  * `:base` - Use 1024 if you wish to format bytes.  Default: 1000
+
+  * `:separator` - The string to place between the scaled number and the
+                   prefix + unit. Perhaps you want a space here. Default: ""
+
+  * `:unit` - The unit of measurement, e.g. "M" for Meters. Default: ""
+
+  * `:precision` - The number of decimal places to include. Default: 2
+
+  * `:trim` - Trim trailing zeros. Default: false
+
```

Comment:
I'd like to see a section here on how to configure defaults for these options. Like [here](https://github.com/danielberkompas/number/blob/564ce20a9788336249f6bff658a83804eabcee79/lib/number/percentage.ex#L30-L38).